diff --git a/PHPUnit/Framework/Comparator/DOMDocument.php b/PHPUnit/Framework/Comparator/DOMDocument.php
index 562b8fbcc..8ffb21cf5 100644
--- a/PHPUnit/Framework/Comparator/DOMDocument.php
+++ b/PHPUnit/Framework/Comparator/DOMDocument.php
@@ -83,7 +83,7 @@ public function accepts($expected, $actual)
      *                           fails. Contains information about the
      *                           specific errors that lead to the failure.
      */
-    public function assertEquals($expected, $actual, $delta = 0, $canonicalize = FALSE, $ignoreCase = FALSE)
+    public function assertEquals($expected, $actual, $delta = 0, $canonicalize = FALSE, $ignoreCase = FALSE, array &$processed = array())
     {
         if ($expected->C14N() !== $actual->C14N()) {
             throw new PHPUnit_Framework_ComparisonFailure(
diff --git a/PHPUnit/TextUI/Command.php b/PHPUnit/TextUI/Command.php
index 6cf867014..e050809e0 100644
--- a/PHPUnit/TextUI/Command.php
+++ b/PHPUnit/TextUI/Command.php
@@ -274,7 +274,7 @@ protected function handleArguments(array $argv)
                           'tokenizer', 'No code coverage will be generated.'
                         );
 
-                        continue;
+                        continue 2;
                     }
 
                     if (!extension_loaded('xdebug')) {
@@ -282,7 +282,7 @@ protected function handleArguments(array $argv)
                           'Xdebug', 'No code coverage will be generated.'
                         );
 
-                        continue;
+                        continue 2;
                     }
 
                     switch ($option[0]) {
diff --git a/PHPUnit/Util/Getopt.php b/PHPUnit/Util/Getopt.php
index 41ec26c02..cc0e359fc 100644
--- a/PHPUnit/Util/Getopt.php
+++ b/PHPUnit/Util/Getopt.php
@@ -77,7 +77,11 @@ public static function getopt(array $args, $short_options, $long_options = NULL)
         reset($args);
         array_map('trim', $args);
 
-        while (list($i, $arg) = each($args)) {
+        while (key($args) !== null) {
+            $i  = (int) key($args);
+            $arg = current($args);
+            next($args);
+
             if ($arg == '') {
                 continue;
             }
@@ -136,7 +140,8 @@ protected static function parseShortOption($arg, $short_options, &$opts, &$args)
                         break;
                     }
 
-                    else if (list(, $opt_arg) = each($args)) {
+                    else if ($opt_arg = current($args)) {
+                        next($args);
                     }
 
                     else {
@@ -184,7 +189,8 @@ protected static function parseLongOption($arg, $long_options, &$opts, &$args)
             if (substr($long_opt, -1) == '=') {
                 if (substr($long_opt, -2) != '==') {
                     if (!strlen($opt_arg) &&
-                        !(list(, $opt_arg) = each($args))) {
+                        !($opt_arg = current($args))) {
+                        next($args);
                         throw new PHPUnit_Framework_Exception(
                           "option --$opt requires an argument"
                         );
diff --git a/composer.json b/composer.json
index c6077c8c2..631794877 100644
--- a/composer.json
+++ b/composer.json
@@ -41,7 +41,7 @@
     "repositories": [
         {
             "type": "pear",
-            "url": "http://pear.php.net"
+            "url": "https://pear.php.net"
         }
     ],
     "suggest": {
